### ISSUE TITLE
Publish to npm

### ISSUE THREAD

**Issue by elliottsj** [tapmodo/Jcrop#200]
It would be awesome if Jcrop was officially on npm :grinning:

**Comment by tapmodo** [comment]
I would like to do this. I've published nodejs packages on npm before. I'm just not familiar with how a front-end jQuery package should be structured for npm. If anyone has suggestions (links that show how) I'm definitely game.

**Comment by Manuel-S** [comment]
I would suggest bower instead, as it is the alternative to npm meant for frontend packages.

Here's a short example how to package and publish a jQuery plugin using bower:
https://ksylvest.com/posts/2013-07-23/packaging-a-jquery-plugin-with-bower

**Comment by tapmodo** [comment]
Yes, Jcrop is already available via bower. While I'm familiar with publishing nodejs packages for npm, I don't know any best practices for publishing front-end packages like Jcrop.

**Comment by elliottsj** [comment]
@tapmodo I good convention I've seen is what Facebook uses for its JS libraries:
- `src/`: Source JS files; can be ES6/CoffeeScript/TypeScript/etc.
- `lib/`: CommonJS, ES5-compatible files, compiled from `src/`
- `dist/`: Browser-compatible files (can be used in `<script src="..."></script>`), usually bundled as a [UMD module](https://github.com/umdjs/umd)

e.g. [react](https://github.com/facebook/react), [immutable-js](https://github.com/facebook/immutable-js/)

In the case of Jcrop, the source file _jquery.Jcrop.js_ is already browser-compatible, so all you really need is a _package.json_ with `"main": "js/jquery.Jcrop.js"`: https://github.com/EventMobi/Jcrop/commit/00e1377246859beaf6fd1c3fab2301593b778407.

It looks like @akalinovski has already published a [jcrop npm package](https://www.npmjs.com/package/jcrop). @akalinovski: Can you [transfer ownership](https://docs.npmjs.com/cli/owner) of the jcrop npm package to @tapmodo?

**Comment by tapmodo** [comment]
Thanks for the information. I'll see what I can do. I know packages can be namespaced, but it would be nice if @akalinovski will work with me to allow an official npm release with the jcrop name.

**Comment by akalinovski** [comment]
@tapmodo I gave you access rights to the npm package.

**Comment by tapmodo** [comment]
@akalinovski dude, thank you! i kind of wondered how that showed up on my npm packages. you are awesome! i will hopefully get everything all packaged up here soon.

**Comment by andreitudor** [comment]
Is the package updated already on npm?

**Comment by ismcagdas** [comment]
Hi,

Can you add v2.x to npm as well ?

**Comment by haberbyte** [comment]
Hey @tapmodo,

will the v2 version be published to npm any time soon?
Would love to see this published 😍

**Comment by wisyr** [comment]
Almost 4 years have passed, any progress?
